test/junit/scala/collection/immutable/ArraySeqTest.scala

```diff
@@ -80,4 +80,9 @@ class ArraySeqTest {
   }
   
   @Test def checkSearch: Unit = SeqTests.checkSearch(ArraySeq(0 to 1000: _*), 15,  implicitly[Ordering[Int]])
+
+  @Test
+  def testCooperativeEquality(): Unit = {
+    assertEquals(ArraySeq(1, 2, 3), ArraySeq(1L, 2L, 3L))
```

Comment:
I think we should add more tests because `ArraySeq(1, 2, 3)` is `ArraySeq.ofInt` but `ArraySeq(1, 2) :+ 3` is `ArraySeq.ofRef`.

Reply:
@xuwei-k oops, I missed that. good point

Reply:
Thanks!